hibernate-vector/src/main/java/org/hibernate/vector/OracleVectorArgumentValidator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.vector;
+
+import java.util.List;
+
+import org.hibernate.metamodel.model.domain.DomainType;
+import org.hibernate.query.sqm.SqmExpressible;
+import org.hibernate.query.sqm.produce.function.ArgumentsValidator;
+import org.hibernate.query.sqm.produce.function.FunctionArgumentException;
+import org.hibernate.query.sqm.tree.SqmTypedNode;
+import org.hibernate.type.BasicType;
+import org.hibernate.type.SqlTypes;
+import org.hibernate.type.spi.TypeConfiguration;
+
+/**
+ * A {@link ArgumentsValidator} that validates the arguments are all 4 oracle vector types i.e. {@link SqlTypes#VECTOR}, {@link SqlTypes#VECTOR_FLOAT32}, {@link SqlTypes#VECTOR_FLOAT64} and {@link SqlTypes#VECTOR_INT8}.
+ */
+public class OracleVectorArgumentValidator implements ArgumentsValidator {
```

Comment:
I would prefer if you alter `VectorArgumentValidator` to support the additional type codes instead of adding a new validator.